app/services/grant_services/new.rb

```diff
@@ -3,23 +3,21 @@
 module GrantServices
   module New
     def self.call(grant:, user:)
-      begin
-        # TODO: is requires_new needed?
-        ActiveRecord::Base.transaction(requires_new: true) do
-          grant.save!
-          # Add user as admin
-          GrantPermission.create!(grant: grant, user: user, role: 'admin')
-          # Create a starter form
-          GrantSubmissionFormServices::New.call(grant: grant, user: user)
-          # Create starter criteria
-          CriterionServices::New.call(grant: grant)
-        end
-        OpenStruct.new(success?: true)
-      rescue
-        #TODO: Log errors
-        OpenStruct.new(success?: false,
-                       messages: grant.errors.any? ? grant.errors.full_messages : 'An error occurred while saving this grant. Please try again.')
+      # TODO: is requires_new needed?
+      ActiveRecord::Base.transaction(requires_new: true) do
+        grant.save!
+        # Add user as admin
+        GrantPermission.create!(grant: grant, user: user, role: 'admin')
+        # Create a starter form
+        GrantSubmissionFormServices::New.call(grant: grant, user: user)
+        # Create starter criteria
+        CriterionServices::New.call(grant: grant)
       end
+      OpenStruct.new(success?: true)
+    rescue
```

Comment:
the tests look pretty good but these `rescue` all exception usages need some review. It seems that you are calling a series of internal services which should not throw any unexpected exceptions unless there's bug in the code, in which case you should let the app fail and trigger an alert instead of hiding it. This can be done as another ticket but I see this as a more urgent technical debt.

Reply:
we can go over this in detail on Friday's POD meeting